doc/faq.rst

```diff
@@ -489,9 +489,43 @@ limited to years between :data:`datetime.MINYEAR` (usually 1) and
 driver) can store BSON datetimes with year values far outside those supported
 by :class:`datetime.datetime`.
 
-There are a few ways to work around this issue. One option is to filter
-out documents with values outside of the range supported by
-:class:`datetime.datetime`::
+There are a few ways to work around this issue. Starting with PyMongo 4.3,
+:func:`bson.decode` can decode BSON datetimes in one of four ways, and can
+be specified using the ``datetime_conversion`` parameter of
+:class:`~bson.codec_options.CodecOptions`.
+
+The default option is
+:attr:`~bson.codec_options.DatetimeConversionOpts.DATETIME`, which will
+attempt to decode as a :class:`datetime.datetime`, allowing
+:class:`~builtin.OverflowError` to occur upon out-of-range dates.
+:attr:`~bson.codec_options.DatetimeConversionOpts.DATETIME_AUTO` alters
+this behavior to instead return :class:`~bson.datetime_ms.DatetimeMS` when
+representations are out-of-range, while returning :class:`~datetime.datetime`
+objects as before:
+
+.. doctest::
+
+    >>> from datetime import datetime
+    >>> from bson.datetime_ms import DatetimeMS
+    >>> from bson.codec_options import DatetimeConversionOpts
+    >>> from pymongo import MongoClient
+    >>> client = MongoClient('localhost', 27017, datetime_conversion=DatetimeConversionOpts.DATETIME_AUTO)
+    >>> client.db.collection.insert_one({"x": datetime(1970, 1, 1)})
+    <pymongo.results.InsertOneResult object at 0x1032cf880>
+    >>> client.db.collection.insert_one({"x": DatetimeMS(2**62)})
+    <pymongo.results.InsertOneResult object at 0x1032cf880>
+    >>> for x in client.db.collection.find():
+    ...     print(x)
+    {'_id': ObjectId('62e45f36e7a1bacf393dbf60'), 'x': datetime.datetime(1970, 1, 1, 0, 0)}
+    {'_id': ObjectId('62e45f36e7a1bacf393dbf61'), 'x': DatetimeMS(4611686018427387904)}
+
+For other options, please refer to
+:class:`~bson.codec_options.DatetimeConversionOpts`.
+
+If we need to use the default decoding behavior with datetimes not supported by
```

Comment:
I'd suggest changing from "If we need to use the default decoding behavior ... filter out documents ..." -> "Another option that does not involve setting `datetime_conversion` is to to filter out documents ..."